javav2/example_code/iot/src/main/java/com/example/iot/HelloIoT.java

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.iot;
+
+// snippet-start:[iot.java2.hello_iot.main]
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.ListThingsRequest;
+import software.amazon.awssdk.services.iot.model.ListThingsResponse;
+import software.amazon.awssdk.services.iot.model.ThingAttribute;
+import java.util.List;
+
+public class HelloIoT {
+    public static void main(String[] args) {
+        System.out.println("Hello AWS IoT. Here is a listing of your AWS IoT Things:");
+        IotClient iotClient = IotClient.builder()
+            .region(Region.US_EAST_1)
+            .build();
+
+        listAllThings(iotClient);
+    }
+
+    public static void listAllThings( IotClient iotClient) {
+        ListThingsRequest thingsRequest = ListThingsRequest.builder()
+            .maxResults(10)
+            .build();
+
+        ListThingsResponse response = iotClient.listThings(thingsRequest) ;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.